helm-projectile.el

```diff
@@ -480,34 +476,20 @@ CANDIDATE is the selected file.  Used when no file is explicitly marked."
      . "Add files to Dired buffer `C-c a'"))
   "Action for files.")
 
-(defun helm-projectile-build-dwim-source (candidates)
-  "Dynamically build a Helm source definition for Projectile files based on context with CANDIDATES."
-  ""
-  (helm-build-in-buffer-source "Projectile files"
-    :data candidates
-    :fuzzy-match helm-projectile-fuzzy-match
-    :coerce 'helm-projectile-coerce-file
-    :action-transformer 'helm-find-files-action-transformer
-    :keymap helm-projectile-find-file-map
-    :help-message helm-ff-help-message
-    :mode-line helm-read-file-name-mode-line-string
-    :action helm-projectile-file-actions
-    :persistent-action #'helm-projectile-file-persistent-action))
-
 (defvar helm-source-projectile-files-list
   (helm-build-in-buffer-source "Projectile files"
-    :data (lambda ()
-            (condition-case nil
-                (projectile-current-project-files)
-              (error nil)))
+    :candidates (lambda ()
+                  (condition-case nil
+                      (cl-loop with root = (projectile-project-root)
```

Comment:
I generally dislike `cl-loop` as I think it creates code that's a bit confusing. I'd go with `seq.el` here, but I can live with this.